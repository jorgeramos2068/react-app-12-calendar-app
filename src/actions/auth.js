import Swal from 'sweetalert2';
import { noTokenFetch, tokenFetch } from "../helpers/fetch";
import types from "../types/types";

export const startLogin = (email, password) => {
  return async (dispatch) => {
    const endpoint = 'auth';
    const data = {
      email,
      password
    };
    const resp = await noTokenFetch(endpoint, data, 'POST');
    const body = await resp.json();
    if (body.ok) {
      localStorage.setItem('token', body.token);
      localStorage.setItem('token-init-date', new Date().getTime());
      const user = {
        uid: body.uid,
        name: body.name
      }
      dispatch(login(user));
    }
    else {
      Swal.fire('Error', body.msg, 'error');
    }
  };
};

export const login = (user) => ({
  type: types.authLogin,
  payload: user
});

export const startRegister = (email, password, name) => {
  return async (dispatch) => {
    const endpoint = 'auth/new';
    const data = {
      email,
      password,
      name
    };
    const resp = await noTokenFetch(endpoint, data, 'POST');
    const body = await resp.json();
    if (body.ok) {
      localStorage.setItem('token', body.token);
      localStorage.setItem('token-init-date', new Date().getTime());
      const user = {
        uid: body.uid,
        name: body.name
      }
      dispatch(login(user));
    }
    else {
      Swal.fire('Error', body.msg, 'error');
    }
  };
};

export const startChecking = () => {
  return async (dispatch) => {
    const endpoint = 'auth/renew';
    const resp = await tokenFetch(endpoint, {}, 'GET');
    const body = await resp.json();
    if (body.ok) {
      localStorage.setItem('token', body.token);
      localStorage.setItem('token-init-date', new Date().getTime());
      const user = {
        uid: body.uid,
        name: body.name
      }
      dispatch(login(user));
    }
    else {
      dispatch(finishChecking());
    }
  };
};

export const finishChecking = () => ({
  type: types.authChecking
});

export const startLogout = () => {
  return (dispatch) => {
    localStorage.clear();
    dispatch(logout());
  };
};

export const logout = () => ({
  type: types.authLogout
});
