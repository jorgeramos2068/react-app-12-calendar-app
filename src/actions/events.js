import Swal from 'sweetalert2';
import types from '../types/types';
import { tokenFetch } from '../helpers/fetch';
import { prepareEvents } from '../helpers/prepareEvents';

export const eventStartAddNew = (event) => {
  return async (dispatch, getState) => {
    const {uid, name} = getState().auth;
    try {
      const endpoint = 'events'
      const resp = await tokenFetch(endpoint, event, 'POST');
      const body = await resp.json();
      if (body.ok) {
        event.id = body.event.id;
        event.user = {
          _id: uid,
          name: name
        };
        dispatch(eventAddNew(event));
      }
    } catch (error) {
      console.log(error);
    }
  }
};

export const eventAddNew = (event) => ({
  type: types.eventAddNew,
  payload: event
});

export const eventStartLoadAll = () => {
  return async (dispatch) => {
    try {
      const endpoint = 'events';
      const resp = await tokenFetch(endpoint);
      const body = await resp.json();
      if (body.ok) {
        const events = prepareEvents(body.events);
        dispatch(eventLoadAll(events));
      }
    } catch (error) {
      console.log(error);
    }
  };
};

export const eventLoadAll = (events) => ({
  type: types.eventLoadAll,
  payload: events
});

export const eventSetActive = (event) => ({
  type: types.eventSetActive,
  payload: event
});

export const eventClearActive = () => ({
  type: types.eventClearActive
});

export const eventStartUpdate = (event) => {
  return async (dispatch) => {
    try {
      const endpoint = `events/${event.id}`;
      const resp = await tokenFetch(endpoint, event, 'PUT');
      const body = resp.json();
      if (body.ok) {
        dispatch(eventUpdate(event));
      }
      else {
        Swal.fire('Error', body.msg, 'error');
      }
    } catch (error) {
      console.log(error);
    }
  };
};

export const eventUpdate = (event) => ({
  type: types.eventUpdate,
  payload: event
});

export const eventStartDelete = () => {
  return async (dispatch, getState) => {
    const {id} = getstate().calendar.activeEvent;
    try {
      const endpoint = `events/${id}`;
      const resp = await tokenFetch(endpoint, {}, 'DELETE');
      const body = resp.json();
      if (body.ok) {
        dispatch(eventDelete());
      }
      else {
        Swal.fire('Error', body.msg, 'error');
      }
    } catch (error) {
      console.log(error);
    }
  };
};

export const eventDelete = () => ({
  type: types.eventDelete
});
