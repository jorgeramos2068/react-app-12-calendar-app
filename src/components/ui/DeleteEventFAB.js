import React from 'react';
import { useDispatch } from 'react-redux';
import { eventDelete } from '../../actions/events';

const DeleteEventFAB = () => {
  const dispatch = useDispatch();

  const handleDelete = () => {
    dispatch(eventDelete());
  }

  return (
    <button
      className="btn btn-danger fab-danger"
      onClick={handleDelete}
    >
      <i className="fas fa-trash"></i>
      <span> Delete event</span>
    </button>
  );
};

export default DeleteEventFAB;
