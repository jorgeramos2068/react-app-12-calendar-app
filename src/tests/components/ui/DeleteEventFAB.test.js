import React from 'react';
import { Provider } from 'react-redux'
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import DeleteEventFAB from '../../../components/ui/DeleteEventFAB';
import { eventStartDelete } from '../../../actions/events';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
// Mocks for actions
jest.mock('../../../actions/events', () => ({
  eventStartDelete: jest.fn()
}));

describe('Tests in DeleteEventFAB component', () => {
  let wrapper;
  let store;
  const initialState = {}

  beforeEach(() => {
    jest.clearAllMocks();
    store = mockStore(initialState);
    store.dispatch = jest.fn();
    wrapper = mount(
      <Provider store={store}>
        <DeleteEventFAB />
      </Provider>
    );
  });

  test('Should display DeleteEventFAB component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should dispatch eventStartDelete action', () => {
    wrapper.find('.btn-danger').prop('onClick')();
    expect(eventStartDelete).toHaveBeenCalled();
  });
});
