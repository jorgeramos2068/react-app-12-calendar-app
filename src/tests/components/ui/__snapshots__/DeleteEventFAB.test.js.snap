// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in DeleteEventFAB component Should display DeleteEventFAB component correctly 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
     Delete event
  </span>
</button>
`;
